From 649c78179ad43f9ee99a2fb12edad5723c1d57b8 Mon Sep 17 00:00:00 2001
From: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
Date: Wed, 22 Feb 2017 16:58:02 +0300
Subject: [PATCH 10/17] mic: vop: use vringh_complete_kern_io()

This is preparation for use of VOP in non-MIC environment.

In VOP, used ring is located in i/o space, which causes bus faults
in vringh_complete_kern() on some setups.

Use vringh_complete_kern_io() instead.

Signed-off-by: Nikita Yushchenko <nikita.yoush@cogentembedded.com>
---
 drivers/misc/mic/vop/vop_vringh.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/mic/vop/vop_vringh.c b/drivers/misc/mic/vop/vop_vringh.c
index 4667929..5dbfb8b 100644
--- a/drivers/misc/mic/vop/vop_vringh.c
+++ b/drivers/misc/mic/vop/vop_vringh.c
@@ -809,7 +809,7 @@ static int _vop_virtio_copy(struct vop_vdev *vdev, struct mic_copy_desc *copy)
 		/* Determine the total data consumed */
 		total += vop_vringh_iov_consumed(riov);
 		total += vop_vringh_iov_consumed(wiov);
-		vringh_complete_kern(vrh, *head, total);
+		vringh_complete_kern_io(vrh, *head, total);
 		*head = USHRT_MAX;
 		if (vringh_need_notify_kern(vrh) > 0)
 			vringh_notify(vrh);
-- 
1.9.1

